### PR TITLE
main_loop

### PR DESCRIPTION
**Pull request by eneisen**
As disscussed, some basic changes

### CODE REVIEW

**Comment by dhimiter49** [comment]
Hey,

I found only this slightly ugly way of adding targets to the data_loader
`
    data_w_target = []
    for data in tqdm(data_loader):
        data.append(torch.FloatTensor(data[0].shape[0],).fill_(1.0))
        data_w_target.append(data)

    data_loader = torch.utils.data.DataLoader(
        data_w_target, batch_size=batch_size, shuffle=True, num_workers=1
    )

    data_w_target = []
    for data in tqdm(data_loader_test):
        data.append(torch.FloatTensor(data[0].shape[0],).fill_(1.0))
        data_w_target.append(data)

    data_loader_test = torch.utils.data.DataLoader(
        data_w_target, batch_size=batch_size, shuffle=True, num_workers=1
    )

`
so simply iterate over the data and add ones. This comes before the loop.
Inside the loop we create the z and create a dataloader for it to pass to the generator
`
        z = torch.FloatTensor(np.random.normal(0, 1, (len(data_loader), 128, 4, 4)))
        data_loader_latents = torch.utils.data.DataLoader(
            z, batch_size=test_batch_size, shuffle=False, num_workers=1
        )
`
The the generator must return also all the generated outputs which isn't implemented correctly right now. We need the generated images to update the discriminator.

For updating the discriminator lets pass the data loader with the input images and a new data loader with generated images.

Also uncomment the lines for the Path, maybe try to fix them because in linux it does not give me an error. I also imported tqdm wrong, change it into:
`
from tqdm import tqdm
`
